### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by Defibrilator**
#### Refactor of the java package. 

Separated the different classes into packages : 
- activity
- calendar
- house
- login
- shop
- task
- user
- util

Moved a couple functions but not much change since current classes are not very linked between different packages. Main changes were for the different kinds of messages.

### CODE REVIEW

**Review by aindreias — APPROVED** [review]
This was a much needed refactoring. The changes are intuitive and clean - thank you very much for your work!

File: app/src/main/java/com/github/houseorganizer/houseorganizer/activity/MainScreenActivity.java
Comment (Arkadyus) [inline app/src/main/java/com/github/houseorganizer/houseorganizer/activity/MainScreenActivity.java:241]: Can these methods not be moved in their respective packages ? They dont feel like they belong in MainScreenActivity to me

File: app/src/main/java/com/github/houseorganizer/houseorganizer/util/Util.java
Comment (Arkadyus) [inline app/src/main/java/com/github/houseorganizer/houseorganizer/util/Util.java:65]: We can probably use the same message for the log and the toast, also I dont think putting the log tag and log message in a list makes sense other than for making codeclimate happy, we should talk about that and maybe ignore codeclimate
(or using the same message would also fix it)

File: app/src/main/java/com/github/houseorganizer/houseorganizer/util/Util.java
Comment (Arkadyus) [inline app/src/main/java/com/github/houseorganizer/houseorganizer/util/Util.java:79]: Im also not too sure about extracting this method since its one line and the line is different in every case

**Review by Arkadyus — CHANGES_REQUESTED** [review]
Good refactoring, though i have a few questions (as comments) about it. Also please optimize the imports before merging
Also shouldnt we separate the xml files in packages as well ?

File: app/src/main/java/com/github/houseorganizer/houseorganizer/util/Util.java
Comment (Arkadyus) [inline app/src/main/java/com/github/houseorganizer/houseorganizer/util/Util.java:36]: This method might be able to be moved into the Event class also, since it is about events

File: app/src/main/java/com/github/houseorganizer/houseorganizer/activity/MainScreenActivity.java
Comment (Defibrilator) [reply app/src/main/java/com/github/houseorganizer/houseorganizer/activity/MainScreenActivity.java:241]: Good idea

File: app/src/main/java/com/github/houseorganizer/houseorganizer/util/Util.java
Comment (Defibrilator) [reply app/src/main/java/com/github/houseorganizer/houseorganizer/util/Util.java:65]: I don't think so, log is for the debugger and toast is for the user hence they cannot have the same message. 
We can talk about which type to use on our meeting.

File: app/src/main/java/com/github/houseorganizer/houseorganizer/util/Util.java
Comment (Defibrilator) [reply app/src/main/java/com/github/houseorganizer/houseorganizer/util/Util.java:79]: Yeah I guess its not very useful but it does make the email login methods a tiny bit clearer. It was already extracted to LoginHelpers, I just moved it to Util to make it more generic.

File: app/src/main/java/com/github/houseorganizer/houseorganizer/activity/MainScreenActivity.java
Comment (Defibrilator) [reply app/src/main/java/com/github/houseorganizer/houseorganizer/activity/MainScreenActivity.java]: But apparently Code Climate is not okay with that.

File: app/src/main/java/com/github/houseorganizer/houseorganizer/util/Util.java
Comment (Defibrilator) [reply app/src/main/java/com/github/houseorganizer/houseorganizer/util/Util.java]: True

File: app/src/main/java/com/github/houseorganizer/houseorganizer/activity/MainScreenActivity.java
Comment (aindreias) [reply app/src/main/java/com/github/houseorganizer/houseorganizer/activity/MainScreenActivity.java]: I had a similary problem last week, codeclimate only allows a max of 4 arguments per method. So it's difficult to move things around when you need a lot of outside information. Perhaps we could have some a controller or decorator for the calendar which receives many params from MainScreenActivity & uses those in its methods. Not sure if that's a good solution though

File: app/src/main/java/com/github/houseorganizer/houseorganizer/util/Util.java
Comment (Defibrilator) [reply app/src/main/java/com/github/houseorganizer/houseorganizer/util/Util.java:45]: Okay I separated the list into two strings. I increased code climate limit on args because it made calendar methods fail.

File: app/src/main/java/com/github/houseorganizer/houseorganizer/activity/MainScreenActivity.java
Comment (Defibrilator) [reply app/src/main/java/com/github/houseorganizer/houseorganizer/activity/MainScreenActivity.java]: For now I increased the limit to 7 but we can talk about it during our meeting.

File: app/src/main/java/com/github/houseorganizer/houseorganizer/calendar/Calendar.java
Comment (Arkadyus) [inline app/src/main/java/com/github/houseorganizer/houseorganizer/calendar/Calendar.java:101]: Same thing here for funcAndErrMessage

File: app/src/main/java/com/github/houseorganizer/houseorganizer/activity/MainScreenActivity.java
Comment (Arkadyus) [inline app/src/main/java/com/github/houseorganizer/houseorganizer/activity/MainScreenActivity.java:178]: Since the method is moved into Calendar, you can delete this one and replace the calls in the class by calls to calendar.refreshCalendar, the same applies for the other methods that were moved

File: app/src/main/java/com/github/houseorganizer/houseorganizer/activity/MainScreenActivity.java
Comment (Arkadyus) [inline app/src/main/java/com/github/houseorganizer/houseorganizer/activity/MainScreenActivity.java:183]: Could the other methods not be moved for some reason or do you think they belong here ? Im hesitant with them but feel like they might be able to be moved (initializeTaskList, registerForEventImage, loadHousehold, rotateView and rotateLists)

File: app/src/main/java/com/github/houseorganizer/houseorganizer/calendar/Calendar.java
Comment (Defibrilator) [reply app/src/main/java/com/github/houseorganizer/houseorganizer/calendar/Calendar.java:72]: We would need to increase the arg limit to 8, I dont think its worth it right now

File: app/src/main/java/com/github/houseorganizer/houseorganizer/activity/MainScreenActivity.java
Comment (aindreias) [reply app/src/main/java/com/github/houseorganizer/houseorganizer/activity/MainScreenActivity.java:183]: I think the easier (or more difficult?) question is: what methods should be left in MainScreenActivity?

**Comment by Defibrilator** [comment]
> Why is there an activity package but activities are split among other directories ?

Because activity is not the right name apparently, in this package we want the different panels inside the app so settings main screen, info, etc.. but for example every class inside login package is an activity.

File: app/src/main/java/com/github/houseorganizer/houseorganizer/activity/MainScreenActivity.java
Comment (Defibrilator) [reply app/src/main/java/com/github/houseorganizer/houseorganizer/activity/MainScreenActivity.java:183]: My opinion is that initializeTaskList should be left in mainscreen, 
loadHousehold could be extracted but we would need to create a house class and since this method is only used here with a lot of internal variables I do not think it is beneficial.
I could move registerForEventImage to EventsAdapter but I'm not sure it would be beneficial, since you wrote the method here I'll leave it up to you to chose where and how it should be (in an other branch preferably).
I moved rotateView to the class Calendar.
I do not think rotateList should be moved since it manages both kinds of lists which are in different packages.

File: app/src/main/java/com/github/houseorganizer/houseorganizer/activity/MainScreenActivity.java
Comment (Arkadyus) [reply app/src/main/java/com/github/houseorganizer/houseorganizer/activity/MainScreenActivity.java:171]: I think changing line 84 to a lambda makes more sense than keeping the method here personally

File: app/src/main/java/com/github/houseorganizer/houseorganizer/activity/MainScreenActivity.java
Comment (Arkadyus) [reply app/src/main/java/com/github/houseorganizer/houseorganizer/activity/MainScreenActivity.java]: This sounds good to me, ill eventually move registerForEventImage to EventsAdapter most likely (using the cleanup branch)

**Review by Oxydeas — APPROVED** [review]
Renamed the package and two other apporved. Good to me

**Review by Arkadyus — APPROVED** [review]
Looks good to me, if we ever decide on changing some methods' location it should be easier now
Thanks a lot for the refactoring

**Pull request merged**
